### ISSUE TITLE
Feature request: Support for Spring Dependency management plugin

### ISSUE THREAD

**Issue by usr42** [patrikerdes/gradle-use-latest-versions-plugin#9]
Hi @patrikerdes,
me again with another feature request ;)

I'm using the [Spring Dependency management plugin](https://github.com/spring-gradle-plugins/dependency-management-plugin) to control the versions of both the direct and the transitive dependencies of my projects. 

An example (from the official README) looks like:
```
dependencyManagement {
     dependencies {
          dependency 'org.springframework:spring-core:4.0.3.RELEASE'
          dependency group:'commons-logging', name:'commons-logging', version:'1.1.2'
     }
}

dependencies {
     compile 'org.springframework:spring-core'
}
```

More details about the plugin:
https://github.com/spring-gradle-plugins/dependency-management-plugin

Thanks again and cheers,
usr42/Balthasar

**Comment by patrikerdes** [comment]
@usr42 have you tried it? From the example you provided it looks like it would work.

**Comment by usr42** [comment]
I've tried some of the use cases how I use your plugin and found some issues. To make it easier to reproduce I've created a repository on github (every branch should reproduce one issue):
https://github.com/usr42/use-latest-issues

To reproduce the issue run inside the branch:
```
./gradlew useLatestVersions  && ./gradlew useLatestVersionsCheck
```

Currently I've found following issues:
1. https://github.com/usr42/use-latest-issues/tree/spring-dependency-plugin-version-in-gradle-properties-with-quotes: If the version in gradle.properties is **inside of quotes** (both, single (`'`) or double quotes (`"`)) and the Spring dependency plugin is used`./gradlew useLatestVersions` updates the version to `none`.
2. https://github.com/usr42/use-latest-issues/tree/spring-dependency-plugin-version-in-gradle-properties: If the version in gradle.properties is **not inside of quotes** and the Spring dependency plugin is used`./gradlew useLatestVersions` does not update at all.
3. https://github.com/usr42/use-latest-issues/tree/version-in-gradle.properties-without-quotes: If the version in gradle.properties is not inside of quotes and the Spring dependency plugin is **not** used`./gradlew useLatestVersions` does not update at all.
4. https://github.com/usr42/use-latest-issues/tree/version-in-gradle.properties-with-quotes: If the version in gradle.properties is **inside of quotes** (both, single (`'`) or double quotes (`"`)) and the Spring dependency plugin is used`./gradlew useLatestVersions` updates the version correctly. But `./gradlew useLatestVersionsCheck`fails with following error:
```
useLatestVersions failed to update 1 dependency to the latest version:
 - junit:junit ["4.12" -> 4.12]
```

In my projects I'm also using external files I apply to my build.gradle and following annotion with the Spring dependency plugin:
```
dependencyManagement {
        dependencySet(group: 'org.junit.platform', version: junitPlatformVersion) {
            entry 'junit-platform-engine'
            entry 'junit-platform-runner'
            entry 'junit-platform-launcher'
            entry 'junit-platform-console'
            entry 'junit-platform-suite-api'
        }
}
```
I haven't checked if both is working yet. But I will also test that.

I also want to thank you for your extremly fast responses every time. These were the quickest responses I ever got in OSS projects yet. 👍🏼

**Comment by patrikerdes** [comment]
I can confirm the following:
1. The new `gradle.properties` functionality isn't working at all in 0.2.5, it was working at some point before I published it but I must have did some last minute change that broke everything. Shame on me for not having unit tests for it
2. I had not thought of having the version in quotes, so that was never working

I'll focus on fixing these two issues first, and then we'll see if there is anything left to fix to get the spring-dependency-plugin working. But I can't promise that I will get fixed as quick as the previous issues :)

**Comment by patrikerdes** [comment]
The gradle.properties bugs are fixed on `master`: https://github.com/patrikerdes/gradle-use-latest-versions-plugin/commit/388b83a8d704d33ba5c3eefffb4258b8d95aa0aa
I'll release a new version soon

**Comment by patrikerdes** [comment]
Cases 2 and 3, where the version is not in quotes in gradle.properties are now working after the latest fix. 
Cases 1 and 4, where the version is in quotes, does not seem to be a valid use case. At least in case 4, you can see that the dependency resolution fails if you run `./gradlew dependencies`, it will output `junit:junit:"4.0" FAILED`. I removed the possibility to have quotes in variables.

**Comment by patrikerdes** [comment]
Fixed in 0.2.6
